concent_api/core/tests/utils.py

```diff
@@ -299,6 +295,47 @@ def _get_deserialized_task_to_compute(
         )
         return task_to_compute
 
+    def _get_deserialized_want_to_compute_task(
+        self,
+        timestamp: Union[str, datetime.datetime, None] = None,
+        node_name: int = None,
+        task_id: Optional[str] = None,
+        perf_index = None,
+        max_resource_size = None,
+        max_memory_size = None,
+        num_cores = None,
+        price = None,
+        concent_enabled = None,
+        extra_data = None,
+        provider_public_key: Optional[str] = None,
+        provider_ethereum_public_key: Optional[str] = None,
+    ) -> WantToComputeTask:
+
+        """ Returns WantToComputeTask deserialized. """
+        with freeze_time(timestamp or get_timestamp_string()):
+            want_to_compute_task = WantToComputeTaskFactory(
+                node_name=node_name,
+                task_id=task_id,
+                perf_index=perf_index,
+                max_resource_size=max_resource_size,
+                max_memory_size=max_memory_size,
+                num_cores=num_cores,
+                price=price,
+                concent_enabled=concent_enabled,
+                extra_data=extra_data,
+                provider_public_key=(
+                    provider_public_key if provider_public_key is not None else self._get_provider_hex_public_key()
+                ),
+                provider_ethereum_public_key=(
+                    provider_ethereum_public_key if provider_ethereum_public_key is not None else self._get_provider_ethereum_hex_public_key()
+                ),
+            )
```

Comment:
I think, that it should not be provided `None` everywhere. If there is `None` it should get values from Golem's Factories.

Reply:
We don't do it like that in any other field in any other message I don't think so. We need to by able to create `WantToComputeTask` with custom keys. We had it like that in `TaskToCompute` creation, I just move it from one message to another because it moved in GM.